### DISCUSSION TITLE
Synchrotron frequency with harmonic and main cavity in pyat

### DISCUSSION THREAD

**Discussion by oscarxblanco** [atcollab/at#595] [Q&A] [answered]
Dear all,

I have a ring with a main and a harmonic rf cavity from which I would like to have the synchrotron frequency.
In matlab, using atx I get a varying frequency for different timelag settings of the harmonic cavity.
In python, however, the output of print(at.radiation_parameters(ring)), I see no difference if the synchrotron frequency.

Is there any other function in pyat to get the synchrotron frequency ?
Is the output of radiation parameters expected to be different to atx in matlab ?

Best regards,
o

**Comment by swhite2401** [comment, marked as answer]
Dear @oscarxblanco,

I admit this is very confusing in AT but we have to maintain these historical functions for backward compatibility: the results is that there are several ways to get the same quantity with different approximations. 

After a quick look at the code:
The synchrotron frequency in `at.radiation_parameters(ring)` derives from analytical formulas that do not take into account the harmonic voltage. atx on the other hand uses the 6D transfer map that includes everything.

In pyAT you may use ring.get_tune() that can either analyze the 6D map like atx or compute tunes form tbt data, this would be my recommended method to get the tunes.
The Tunes output of print(at.radiation_parameters(ring)) also returns these values if you prefer this method (to get it for 6D -> params = at.radiation_parameters(ring), qs = params.tunes6[2]).

You need to multiply the tune by ring.revolution_frequency to get the synchrotron frequency.

We probably need to document this better, please let me know if this solves your issues.
Cheers

**Reply by oscarxblanco** [reply]
Thank you for the clarification, I'll do some tests.
